crates/pallets/starknet/src/lib.rs

```diff
@@ -826,18 +809,39 @@ impl<T: Config> Pallet<T> {
     fn validate_tx(transaction: Transaction, tx_type: TxType) -> Result<(), TransactionValidityError> {
         let mut state: BlockifierStateAdapter<T> = BlockifierStateAdapter::<T>::default();
         let mut execution_resources = ExecutionResources::default();
-
-        let block_context = Self::current_block()
-            .header()
-            .clone()
-            .into_block_context(Pallet::<T>::fee_token_address(), ChainId(Pallet::<T>::chain_id_str()));
-
+        let block_context = Self::get_block_context();
         transaction
             .validate_account_tx(&mut state, &mut execution_resources, &block_context, &tx_type)
             .map_err(|_err| TransactionValidityError::Invalid(InvalidTransaction::BadProof))?;
 
         Ok(())
     }
+
```

Comment:
add documentation about the rational of this function. it's not a simple getter function, it plays a big role in performance optimization and it should be clear reading the doc of it. Explain the rationale and why it is helpful for perf optim